src/components/content/credential/AddCredential.tsx

```diff
@@ -0,0 +1,248 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+import TextArea from 'antd/es/input/TextArea';
+import { ColumnsType } from 'antd/es/table';
+import { usernameKey } from '../../utils/constants';
+import { Button, Form, FormInstance, Input, Select, Table } from 'antd';
+import React, { ChangeEvent, Ref, useEffect, useState } from 'react';
+import {
+    CreateCredential,
+    CredentialsManagementService,
+    CredentialVariable,
+    CredentialVariables,
+} from '../../../xpanse-api/generated';
+
+function AddCredential({
+    getCredentials,
+    formRef,
+    onCancel,
+    getTipInfo,
+}: {
+    getCredentials: () => void;
+    formRef: Ref<FormInstance<CredentialVariables>> | undefined;
+    onCancel: () => void;
+    getTipInfo: (tipType: 'error' | 'success' | undefined, tipMessage: string) => void;
+}): JSX.Element {
+    const [form] = Form.useForm();
+    const [currentCsp, setCurrentCsp] = useState<CredentialVariables.csp | undefined>(undefined);
+    const [currentType, setCurrentType] = useState<CredentialVariables.type | undefined>(undefined);
+    const [credentialTypeList, setCredentialTypeList] = useState<CredentialVariables.type[]>([]);
+    const [credentialVariableList, setCredentialVariableList] = useState<CredentialVariable[]>([]);
+    const handleCspSelect = (cspName: CredentialVariables.csp) => {
+        setCurrentCsp(cspName);
+    };
+
+    const clear = () => {
+        setCurrentCsp(undefined);
+        setCurrentType(undefined);
+        setCredentialTypeList([]);
+        setCredentialVariableList([]);
+    };
+
+    const handleCredentialTypeSelect = (type: CredentialVariables.type) => {
+        setCurrentType(type);
+        form.setFieldsValue({ variables: [] });
+    };
+
+    useEffect(() => {
+        if (currentCsp === undefined) {
+            return;
+        }
+        void CredentialsManagementService.getCredentialTypesByCsp(currentCsp).then((resp) => {
```

Comment:
We must handle exception here. 
If the API returns 400, then error must be displayed.